cmd/server.go

```diff
@@ -195,15 +195,23 @@ func NewServerCommand(stopCh <-chan struct{}) *cobra.Command {
 					}
 					if err != nil {
 						if etcdErr, ok := err.(*errors.EtcdError); ok == true {
-							logger.Errorf("Failed to take first delta snapshot: snapshotter failed with etcd error: %v", etcdErr)
+							logger.Warnf("Failed to take first delta snapshot: snapshotter failed with etcd error: %v", etcdErr)
 						} else {
-							logger.Errorf("Failed to take first delta snapshot: snapshotter failed with error: %v", err)
+							logger.Warnf("Failed to take first delta snapshot: snapshotter failed with error: %v", err)
+						}
+						startWithFullSnapshot = false
+						if err := ssr.TakeFullSnapshotAndResetTimer(); err != nil {
+							logger.Errorf("Failed to take substitute first full snapshot: %v", err)
+							continue
 						}
-						continue
 					}
 					if err = ssr.TakeDeltaSnapshot(); err != nil {
```

Comment:
Why not a `TakeFullSnapshotAndResetTimer` here and skip the second call on error?